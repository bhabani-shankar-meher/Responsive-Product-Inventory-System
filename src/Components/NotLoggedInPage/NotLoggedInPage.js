
import { Button } from "@mui/material";
import Box from "@mui/material/Box";
import Grid from "@mui/material/Grid";
import Typography from "@mui/material/Typography";
import Divider from "@mui/material/Divider";
import { useNavigate } from "react-router-dom";

const NotLoggedInPage = (props) => {
    const navigate = useNavigate();
    return (
        <Box sx={{ width: "70%" }} mt={5} mb={3} style={{ marginLeft: "15%" }}>
            <Grid
                container
                rowSpacing={3}
                columnSpacing={{ xs: 1, sm: 2, md: 3 }}
            >
                <Grid item xs={12} md={12}>
                    <Typography textAlign="center" variant="h4">
                        Not Logged In
                    </Typography>
                    <Divider sx={{ width: "100%", marginTop: "20px" }} />
                    <Typography textAlign="center" variant="h6">
                        {props.description}
                    </Typography>
                </Grid>
                <Grid item xs={12} md={12} mt={5} style={{ display: "flex", justifyContent: "center" }}>
                    <Button
                        variant="contained"
                        color="primary"
                        style={{ width: "10rem" }}
                        onClick={() => { navigate("/login", { replace: true }) }}
                    >
                        Login
                    </Button>
                    <Button
                        variant="outlined"
                        color="secondary"
                        style={{ width: "10rem", marginLeft: "2rem" }}
                        onClick={() => navigate("/signup", { replace: true })}
                    >
                        Signup
                    </Button>
                </Grid>
            </Grid>
        </Box>
    );
};
export default NotLoggedInPage;