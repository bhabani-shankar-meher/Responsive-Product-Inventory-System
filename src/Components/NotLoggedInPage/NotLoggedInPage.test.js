import NotLoggedInPage from "./NotLoggedInPage";
import { render, screen } from "@testing-library/react";
import { create } from "react-test-renderer";
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

describe("NOTLOGGEDINPAGE SNAPSHOT", () => {
    test("NotLoggedInPage Snapshot Executed Successfully", () => {
        const notLoggedInSnapshot = create(<NotLoggedInPage />);
        expect(notLoggedInSnapshot).toMatchSnapshot();
    });
});
describe("NOTLOGGEDINPAGE TEST", () => {
    test("NotLoggedInPage Heading Test", () => {
        render(<NotLoggedInPage />);
        expect(
            screen.getByRole("heading", {
                name: "Not Logged In",
            })
        ).toBeInTheDocument();
    });
    test("NotLoggedInPage Test_1", () => {
        render(<NotLoggedInPage />);
        expect(
            screen.getByRole("button", {
                name: /login/i,
            })
        ).toBeInTheDocument();
    });
    test("NotLoggedInPage Test_2", () => {
        render(<NotLoggedInPage />);
        expect(
            screen.getByRole("button", {
                name: /signup/i,
            })
        ).toBeInTheDocument();
    });
});