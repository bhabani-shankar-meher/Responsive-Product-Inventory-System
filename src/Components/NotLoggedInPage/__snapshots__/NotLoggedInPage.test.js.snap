// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NOTLOGGEDINPAGE SNAPSHOT NotLoggedInPage Snapshot Executed Successfully 1`] = `
<div
  className="MuiBox-root css-1ahqaq8"
  style={
    Object {
      "marginLeft": "15%",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container css-qwfiov-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
      >
        Not Logged In
      </h4>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-1dh14ht-MuiDivider-root"
      />
      <h6
        className="MuiTypography-root MuiTypography-h6 css-1kgxqm0-MuiTypography-root"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-2l8bkx-MuiGrid-root"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "10rem",
          }
        }
        tabIndex={0}
        type="button"
      >
        Login
      </button>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginLeft": "2rem",
            "width": "10rem",
          }
        }
        tabIndex={0}
        type="button"
      >
        Signup
      </button>
    </div>
  </div>
</div>
`;
